beacon-chain/casper/validator_test.go

```diff
@@ -384,3 +482,52 @@ func TestMinEmptyValidator(t *testing.T) {
 		t.Errorf("Min vaidator index should be -1")
 	}
 }
+
+func TestDeepCopyValidators(t *testing.T) {
+	var validators []*pb.ValidatorRecord
+	defaultValidator := &pb.ValidatorRecord{
+		Pubkey:            []byte{'k', 'e', 'y'},
+		WithdrawalShard:   2,
+		WithdrawalAddress: []byte{'a', 'd', 'd', 'r', 'e', 's', 's'},
+		RandaoCommitment:  []byte{'r', 'a', 'n', 'd', 'a', 'o'},
+		Balance:           uint64(1e9),
+		Status:            uint64(params.Active),
+		ExitSlot:          10,
+	}
+	for i := 0; i < 100; i++ {
+		validators = append(validators, defaultValidator)
+	}
+
+	newValidatorSet := DeepCopyValidators(validators)
```

Comment:
I think a better test would modify the original validator set and verify that it doesn't modify the new one :)